sklearn/cluster/_optics.py

```diff
@@ -233,7 +233,7 @@ class OPTICS(ClusterMixin, BaseEstimator):
     _parameter_constraints: dict = {
         "min_samples": [
             Interval(Integral, 2, None, closed="left"),
-            Interval(Real, 0, 1, closed="both"),
+            Interval("real_not_int", 0, 1, closed="both"),
```

Comment:
Naive question: do you think it would be possible to implement `ReadNotInt` which would extend [`Number`](https://docs.python.org/3/library/numbers.html#numbers.Number) or even [`Real`](https://docs.python.org/3/library/numbers.html#numbers.Real)and perfom the check?

Reply:
It's doable and I did it in this commit [718523f](https://github.com/scikit-learn/scikit-learn/pull/25797/commits/718523f905129dad9b26368e8f2ddf7a404d9f44) to show how it would look like.

I don't have a strong opinion about using this instead of 'real_not_int'. I slightly tend toward using ``RealNotInt``. Do you have an opinion @glemaitre @thomasjpfan ?

Reply:
I'm okay with `RealNotInt`. It looks a little better when placed next to `Integral`.

Reply:
Good with me as well.